datafusion/core/tests/sql/sql_api.rs

```diff
@@ -39,6 +52,99 @@ async fn unsupported_ddl_returns_error() {
     ctx.sql_with_options(sql, options).await.unwrap();
 }
 
+struct MockFunctionFactory;
```

Comment:
This is very cool -- I think it would be great (as a follow on PR) if we turned this into an example (with docs, etc like the others in https://github.com/apache/arrow-datafusion/tree/main/datafusion-examples/examples)

Reply:
Right, this functionality looks important and definitely needs to be covered with docs, examples in following PR, looking forward for it